Update MSRV to one that supports newer rusqlite, cbindgen

https://github.com/GothenburgBitFactory/taskwarrior/pull/2788
```
error[E0658]: const generics are unstable
```

https://github.com/GothenburgBitFactory/taskwarrior/pull/2789
```
error[E0658]: the `unsafe_op_in_unsafe_fn` lint is unstable
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/os_str_bytes-6.0.1/src/lib.rs:159:1
    |
159 | #![warn(unsafe_op_in_unsafe_fn)]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #71668 <https://github.com/rust-lang/rust/issues/71668> for more information

error[E0658]: use of unstable library feature 'slice_strip'
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/os_str_bytes-6.0.1/src/raw_str.rs:606:16
    |
606 |         self.0.strip_prefix(pat).map(Self::from_raw_bytes_unchecked)
    |                ^^^^^^^^^^^^
    |
    = note: see issue #73413 <https://github.com/rust-lang/rust/issues/73413> for more information
```

/cc @dbr

--- a/.github/workflows/rust-tests.yml
+++ b/.github/workflows/rust-tests.yml
@@ -13,7 +13,7 @@ jobs:
       matrix:
         rust:
           # MSRV; most not be higher than the clippy rust version in checks.yml
-          - "1.47"
+          - "1.51"
           - "stable"
         os:
           - ubuntu-latest

--- a/.github/workflows/rust-tests.yml
+++ b/.github/workflows/rust-tests.yml
@@ -13,7 +13,7 @@ jobs:
       matrix:
         rust:
           # MSRV; most not be higher than the clippy rust version in checks.yml
-          - "1.51"
+          - "1.52"
           - "stable"
         os:
           - ubuntu-latest

--- a/rust/lib/src/lib.rs
+++ b/rust/lib/src/lib.rs
@@ -1,5 +1,4 @@
-// Not compatible with the MSRV
-// #![warn(unsafe_op_in_unsafe_fn)]
+#![warn(unsafe_op_in_unsafe_fn)]
 #![allow(unused_unsafe)]
 // Not working yet in stable - https://github.com/rust-lang/rust-clippy/issues/8020
 // #![warn(clippy::undocumented_unsafe_blocks)]

--- a/rust/taskchampion/src/lib.rs
+++ b/rust/taskchampion/src/lib.rs
@@ -41,7 +41,7 @@ for more information about the design and usage of the tool.
 
 # Minimum Supported Rust Version (MSRV)
 
-This crate supports Rust version 1.51 and higher.
+This crate supports Rust version 1.52 and higher.
 
  */
 

--- a/rust/taskchampion/src/lib.rs
+++ b/rust/taskchampion/src/lib.rs
@@ -39,9 +39,9 @@ Users can define their own server impelementations.
 See the [TaskChampion Book](http://taskchampion.github.com/taskchampion)
 for more information about the design and usage of the tool.
 
-# Minimum Supported Rust Version
+# Minimum Supported Rust Version (MSRV)
 
-This crate supports Rust version 1.47 and higher.
+This crate supports Rust version 1.51 and higher.
 
  */
 

